core/src/worker/data-feed.utils.ts

```diff
@@ -52,16 +52,18 @@ export async function oracleRoundStateCall({
   oracleAddress,
   operatorAddress,
   logger,
-  roundId
+  roundId,
+  provider
 }: {
   oracleAddress: string
   operatorAddress: string
   roundId?: number
   logger?: Logger
+  provider?: ethers.providers.JsonRpcProvider
 }): Promise<IOracleRoundState> {
   logger?.debug({ oracleAddress, operatorAddress }, 'oracleRoundStateCall')
-
-  const aggregator = new ethers.Contract(oracleAddress, Aggregator__factory.abi, PROVIDER)
+  if (!provider) provider = PROVIDER
```

Comment:
I think it is safer to make `provider` mandatory parameter and force whoever uses it to pass `JsonRpcProvider`. From the perspective of caller, it is hard to assume what happens when provider is not specified. By making sure that the provider is passed we can avoid such issue. What do you think?

Reply:
Yes, that makes sense since we have L1 and L2 provider now.